contracts/uniswap_exchange.vy

```diff
@@ -534,3 +539,13 @@ def adjust_platform_fee_max(_new_platform_fee_max : uint256) -> bool:
       assert _new_platform_fee_max < self.platform_fee_max
       self.platform_fee_max = _new_platform_fee_max
       return True
+
+@public
+def token_scrape(deadline: timestamp) -> uint256(wei):
+      assert msg.sender == self.owner
+      token_in_contract: uint256 = self.token.balanceOf(self)
+      eth_min_amount: uint256(wei) = self.getTokenToEthInputPrice(token_in_contract)
+      eth_amount: uint256(wei) = self.tokenToEthTransferInput(token_in_contract, eth_min_amount, deadline, self.owner)
+      exchange_addr: address = self.factory.getExchange(self.owner)
+      Exchange(exchange_addr).addLiquidity(1, token_in_contract, deadline)
```

Comment:
Not sure if this is the correct call here, but maybe I'm not familiar enough with vyper